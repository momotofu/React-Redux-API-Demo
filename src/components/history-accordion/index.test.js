import React from 'react'
import { shallow, mount, render } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import jsdom from 'jsdom'

import Accordion from './index'

const doc = jsdom.jsdom('<!doctype html><html><body></body></html>')
global.document = doc
global.window = doc.defaultView

var items =
[{
  "open": false,
  "itemID": 'a9asf98h98ahsfoi13',
  "buildID": "Tenrox-R1_1235",
  "itemState": "pending",
  "owner": "samy",
  "timeStarted": 1478128607000,
  "percentageComplete": 77.1,
  "metrics": {
  "test": 64,
  "maintainability": 53,
  "security": 64,
  "workmanship": 72,
  "isCompleted": true
  },
  "build": {
    "timeStamp": 1478128607000,
    "isCompleted": true
  },
  "unitTest": {
    "passed": 142,
    "failed": 10,
    "coveredPercentage": 76,
    "testsPassed": 73,
    "isCompleted": true
  },
  "functionalTest": {
    "passed": 142,
    "failed": 10,
    "coveredPercentage": 76,
    "testsPassed": 68,
    "isCompleted": true
  },
  "result": {
    "status": "find_issue"
  }
},
{
  "open": false,
  "itemID": 'a9asf9asdf1398ahsfoi13',
  "buildID": "Tenrox-R1_1235",
  "itemState": "pending",
  "owner": "samy",
  "timeStarted": 1478128607000,
  "percentageComplete": 77.1,
  "metrics": {
  "test": 64,
  "maintainability": 53,
  "security": 64,
  "workmanship": 72,
  "isCompleted": true
  },
  "build": {
    "timeStamp": 1478128607000,
    "isCompleted": true
  },
  "unitTest": {
    "passed": 142,
    "failed": 10,
    "coveredPercentage": 76,
    "testsPassed": 73,
    "isCompleted": true
  },
  "functionalTest": {
    "passed": 142,
    "failed": 10,
    "coveredPercentage": 76,
    "testsPassed": 68,
    "isCompleted": true
  },
  "result": {
    "status": "find_issue"
  }
}]

function handleItemClick(id) {
  return
}


describe('Accordion component', function() {
  it('Should match snapshot', function() {
    const wrapper = shallow(<Accordion
                              accordionItems={items}
                              onItemClick={handleItemClick}
                            />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('Allows props to be set', () => {
    const wrapper = mount(<Accordion
                            accordionItems={items}
                            onItemClick={handleItemClick}
                            />)
    expect(wrapper.props().accordionItems[1].functionalTest).toEqual({
      "passed": 142,
      "failed": 10,
      "coveredPercentage": 76,
      "testsPassed": 68,
      "isCompleted": true
    })

    wrapper.setProps({ bar: 'foo' })
    expect(wrapper.props().bar).toEqual('foo')
  })
})