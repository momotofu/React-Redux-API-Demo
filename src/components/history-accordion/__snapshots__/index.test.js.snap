exports[`Accordion component Should match snapshot 1`] = `
<div
  className="Accordion-container">
  <div
    className="Accordion-container-labels">
    <span>
      Changelist/build
    </span>
    <div
      className="Accordion-container-labels-metrics">
      <span
        className="Accordion-container-labels-metrics-label">
        State
      </span>
      <span
        className="Accordion-container-labels-metrics-label">
        Metrics
      </span>
      <span
        className="Accordion-container-labels-metrics-label">
        Build
      </span>
      <span
        className="Accordion-container-labels-metrics-label">
        Unit Test
      </span>
      <span
        className="Accordion-container-labels-metrics-label">
        Func Test
      </span>
    </div>
  </div>
  <AccordionItem
    build={
      Object {
        "isCompleted": true,
        "timeStamp": 1478128607000
      }
    }
    buildID="Tenrox-R1_1235"
    functionalTest={
      Object {
        "coveredPercentage": 76,
        "failed": 10,
        "isCompleted": true,
        "passed": 142,
        "testsPassed": 68
      }
    }
    itemID="a9asf98h98ahsfoi13"
    itemState="pending"
    metrics={
      Object {
        "isCompleted": true,
        "maintainability": 53,
        "security": 64,
        "test": 64,
        "workmanship": 72
      }
    }
    onClick={[Function onClick]}
    open={false}
    owner="samy"
    percentageComplete={77.1}
    result={
      Object {
        "status": "find_issue"
      }
    }
    timeStarted={1478128607000}
    unitTest={
      Object {
        "coveredPercentage": 76,
        "failed": 10,
        "isCompleted": true,
        "passed": 142,
        "testsPassed": 73
      }
    } />
  <AccordionItem
    build={
      Object {
        "isCompleted": true,
        "timeStamp": 1478128607000
      }
    }
    buildID="Tenrox-R1_1235"
    functionalTest={
      Object {
        "coveredPercentage": 76,
        "failed": 10,
        "isCompleted": true,
        "passed": 142,
        "testsPassed": 68
      }
    }
    itemID="a9asf9asdf1398ahsfoi13"
    itemState="pending"
    metrics={
      Object {
        "isCompleted": true,
        "maintainability": 53,
        "security": 64,
        "test": 64,
        "workmanship": 72
      }
    }
    onClick={[Function onClick]}
    open={false}
    owner="samy"
    percentageComplete={77.1}
    result={
      Object {
        "status": "find_issue"
      }
    }
    timeStarted={1478128607000}
    unitTest={
      Object {
        "coveredPercentage": 76,
        "failed": 10,
        "isCompleted": true,
        "passed": 142,
        "testsPassed": 73
      }
    } />
</div>
`;
